its/src/test/java/com/sonar/it/scanner/msbuild/utils/BuildCommand.java

```diff
@@ -56,28 +69,65 @@ public BuildCommand addArgument(String... arguments) {
     return this;
   }
 
-  public BuildResult execute(Orchestrator orchestrator) {
-    // ToDo: SCAN4NET-312: Extract createCommand, add support for parameters, add support for MsBuild vs dotnet. Should be similar to ScannerCommand.execute()
-    // ToDo: SCAN4NET-312: Run restore & rebuild. Restore could be via NuGet (on request, or every time for simplicity?), MsBuild or dotnet
-    var environmentVariables = environment.entrySet().stream().map(x -> new EnvironmentVariable(x.getKey(), x.getValue())).toList();
-    var result = dotnetCommand == null
-      ? TestUtils.runMSBuild(orchestrator, projectDir, environmentVariables, timeout, prependArgument("/t:Restore,Rebuild"))
-      : TestUtils.runDotnetCommand(projectDir, environmentVariables, dotnetCommand, dotnetArguments());
+  public BuildResult execute() {
+    addDefaultArguments();
+    var command = createCommand();
+    var result = new BuildResult();
+    LOG.info("Command line: {}", command.toCommandLine());
+    result.addStatus(CommandExecutor.create().execute(command, new StreamConsumer.Pipe(result.getLogsWriter()), timeout));
     assertTrue(result.isSuccess());
     return result;
   }
 
-  private String[] dotnetArguments() {
-    return dotnetCommand == "build"
-      ? prependArgument("--no-incremental")
-      : arguments.toArray(new String[0]);
+  private void addDefaultArguments() {
+    // If the provided command is not the default one, we assume the arguments are already set
+    if (dotnetCommand != null && !DEFAULT_DOTNET_COMMAND.equals(dotnetCommand)) {
+      return;
+    }
+    // Adding default build arguments
+    // Assuming that whenever we use MSBuild we always want to restore and rebuild
+    arguments.add(0, "-nodeReuse:false");
+    arguments.add(0, "/t:Restore,Rebuild");
+  }
+
+  private Command createCommand() {
+    determineBuildCommand();
+    Command command;
+    if (dotnetCommand == null) {
```

Comment:
There are generally too many `dotnetCommand` null checks. There should be only one, to make it really clear when the decision is taken and what it really does.